pilot/pkg/config/clusterregistry/types.go

```diff
@@ -0,0 +1,143 @@
+/*
```

Comment:
I really dislike `types.go` files - I think it's a pretty bad pattern if types don't naturally fit into other declarations (we only have one other `types.go` in Pilot, and it's automatically generated, and used to create CRDs).

Using the k8s gen-client stuff, we should be able to merge type declarations with the the code that uses them.

Reply:
This file is actually copied from https://github.com/kubernetes/cluster-registry/blob/master/pkg/apis/clusterregistry/types.go since clusterregistry can't be directly imported here due to compile errors (caused by incompatible package requirement between istio and cluster registry). 

The long-term goal is to use clusterregistry, and that's why it's copied here for the time being.

Reply:
Ah, gotcha. Thanks.